pkg/api/api/client.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"net/http"
+
+	"go.jetpack.io/pkg/auth/session"
+)
+
+// Client manages state for interacting with the JetCloud API, as well as
+// communicating with the JetCloud API.
+type Client struct {
+	Host       string
+	httpClient *http.Client
+}
+
+func NewClient(host string, token *session.Token) *Client {
+	return &Client{
+		Host:       host,
+		httpClient: &http.Client{Transport: &transport{token}},
+	}
+}
+
+type transport struct{ token *session.Token }
```

Comment:
I don't think we should define our own transport. Why doesn't `oauth2.NewClient()` or `oauth2.Config.Client()` (for the first one, we could just pass a static token source if all you want to do is pass the token we already have, eventually, if we setup the config and create the client from there, it would handle a token refresh automatically, but we don't need to do that yet)